core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
> Since in Go there is no way to use a value as a key that does not have an '==' operator there is never an issue.

You chose to apply your behavior to floats uniformly by introducing a Comparator that behaves consistently, Go is clever about types and can figure out where floats on its own without additional runtime cost (it's already the) and just call the relevant internal function for that.

This whole thing is about how the hash function treats floats *internally* versus how the *hardware* comparison works. It's about the nature of the behavior, not which language feature you use to override the hardware comparison.